### PR TITLE
First draft of UART communication

### PR DESCRIPTION
**Pull request by murgui**
At the moment, the system is able to print strings:

UARTStringPrint("NicolasIsTheMVP");

and ints:

UARTIntPrint("myVariableName ", (int)(currentPhaseA * 100))

The function only accepts int16, if decimal values are desired to be used, do some conversion like the shown in the previous function. An interface to print floats will be added in the next iteration.

### CODE REVIEW

File: Software/Development/SemesterProject/App/Include/queueObject.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include <limits.h>
+#include <string.h>
+#include "DSP28x_Project.h"
+
+
+
+#define getName(var)  #var
+
+//structure to represent a queue
+struct Queue
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/App/Include/queueObject.h:22]: I had no idea it was spelled like this, weird!

**Review by NicolaiFransen — APPROVED** [review]
¿Ship it?

File: Software/Development/SemesterProject/System/systemInit.c [hunk 2]
```diff
@@ -148,9 +150,12 @@ void systemInit(void)
     initPWM();
     initAnalogSignals();      // Initialize the analog signals and their ADC channels
     initializeGUIPushbuttonsStructure();
-	  initDigitalOutputs();
-	  initPushbuttons();
+    initDigitalOutputs();
+    initPushbuttons();
     initWatchdog();
+    UARTinit();
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/System/systemInit.c]: better to change it to initUART() just to keep format

File: Software/Development/SemesterProject/System/systemInit.c [hunk 3]
```diff
@@ -148,9 +150,12 @@ void systemInit(void)
...
+    UARTStringPrint("\n\rWelcome to the excellent Go-Kart controller, by PED-841.\n\r");
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/System/systemInit.c]: why not including this in UARTinit(), or even better in initUART() ¿?

**Review by AitorTeran — APPROVED** [review]
Looks like some pro code, good... 
Minor comments, almost ready to ship!

File: Software/Development/SemesterProject/System/systemInit.c (hunk 2 above)
Comment (murgui) [reply Software/Development/SemesterProject/System/systemInit.c]: done

File: Software/Development/SemesterProject/System/systemInit.c (hunk 3 above)
Comment (murgui) [reply Software/Development/SemesterProject/System/systemInit.c]: Done

**Pull request merged**